src/components/MAPI/utils.ts

```diff
@@ -444,6 +444,7 @@ export async function fetchControlPlaneNodesForCluster(
               [capzv1alpha3.labelControlPlane]: 'true',
             },
           },
+          namespace: cluster.metadata.namespace,
```

Comment:
Changing the requests for fetching control plane nodes so that it's namespaced: we look for control plane nodes for a cluster in the same namespace as the cluster.